automatminer/automl/keras_wrapper.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import unicode_literals
+import keras.models
+import keras.legacy.layers
+import keras.regularizers
+import sklearn.metrics
+import sklearn.base
+import keras.constraints
+import keras.layers.noise
+import keras.optimizers
+import keras.callbacks
+from sklearn.datasets import load_boston
+from sklearn.model_selection import KFold, cross_val_score
+from keras.wrappers.scikit_learn import KerasRegressor, KerasClassifier
+import pandas as pd
+from gen_evolve_v2 import NeuralNetOptimizer
+import numpy as np
+import random
+
+_classifier_modes = {'classifier', 'classification', 'classify'}
+
+_regressor_modes = {'regressor', 'regression', 'regress'}
+
+class NnWrapper(sklearn.base.BaseEstimator, sklearn.base.RegressorMixin, sklearn.base.ClassifierMixin):
+    """Wrapper for Keras feed-forward neural network for classification to enable scikit-learn grid search"""
+    def __init__(self, init="glorot_uniform", optimizer="adam", hidden_layer_sizes=2, units=20, dropout=0.5, show_accuracy=True, batch_spec=((400, 1024), (100, -1)), activation="sigmoid", input_noise=0., use_maxout=False, use_maxnorm=False, learning_rate=0.001, stop_early=False, kfold_splits = 2, mode="regression"):
+        self.hidden_layer_sizes = hidden_layer_sizes
+        self.init = init
+        self.optimizer = optimizer
+        self.dropout = dropout
+        self.show_accuracy = show_accuracy
+        self.batch_spec = batch_spec
+        self.activation = activation
+        self.input_noise = input_noise
+        self.use_maxout = use_maxout
+        self.use_maxnorm = use_maxnorm
+        self.learning_rate = learning_rate
+        self.stop_early = stop_early
+        self.units = units
+
+        if self.use_maxout:
+            self.use_maxnorm = True
+
+        self.model_ = None
+        self.estimator = None
+        self.classifier = None
+        self.kfold_splits = kfold_splits
+        if mode in _classifier_modes:
+            self.mode = "classification"
+        else:
+            self.mode = "regression"
+        self.fitted_pipeline_ = None
+
+    def getModel(self):
+        return self.model_
```

Comment:
Trailing underscores in python are used to avoid conflicts with keywords (see https://dbader.org/blog/meaning-of-underscores-in-python). If you want this to be private, use self._model.